ls1intum/Artemis
# `Assessment`: Add a dropdown menu for structured grading instructions

### Checklist
#### General


#### Client

### Motivation and Context


So far, the only way to use feedback from SGI is via drag-and-drop. However, this can lead to certain problems. Therefore, tutors now have the choice to also select the grading instructions via a drop-down menu.

### Description

The grading criteria of the exercise are forwarded to the feedback editor.  First the titles of the criteria are displayed in the drop-down menu, hovering brings up sub-menus that contain the instructions corresponding to the respective criterion. These are provided by a new additional component `textblock-feedback-dropdown`.

Some edge cases:
- if there is a criterion without a title: ‘(No title)’ is displayed
- if there is only one criterion: the instructions are displayed immediately within the dropdown menu
- if there are no criteria at all, the dropdown does not appear at all

### Steps for Testing

Prerequisites:

- 1 Tutor
- 1 Student submission of a text-exercise with SGI

1. Navigate to the Exercise Dashboard
2. Start the assessment
3. Select a text block and click on the button to see the dropdown.
4. If there is more than one criterion:
    4.1. Hover over the criterion title and click on the specific instruction.
    4.2. The text block should be marked with the feedback of the chosen instruction and the points should be calculated correctly.
5. If there is only one criterion:
    5.1. Check that there is no submenu, and instructions are shown immediately.

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots



https://user-images.githubusercontent.com/73841445/159497615-3305e731-47f3-4f35-b362-20447a453472.mov


If there is only one criterion:

<img width="865" alt="Bildschirmfoto 2022-04-24 um 16 18 29" src="https://user-images.githubusercontent.com/73841445/165055475-8c99020a-c064-4e65-8a76-6c063917b2ca.png">
